include/glob.hpp

```diff
@@ -28,47 +29,82 @@ namespace glob {
 
 		/* Add */
 		vect operator+(const vect &v) const { return vect(x + v.x,
-		                                                  y + v.y); };
+														  y + v.y); };
 
 		/* Subtract */
 		vect operator-(const vect &v) const { return vect(x - v.x,
-		                                                  y - v.y); };
+														  y - v.y); };
 
 		/* Multiply by a scalar */
 		vect operator*(const float &s) const { return vect(x * s,
-		                                                   y * s); };
+														   y * s); };
 
 		/* Divide by a scalar */
 		vect operator/(const float &s) const { return vect(x / s,
-		                                                   y / s); };
+														   y / s); };
 
 		/* Dot product */
-		float operator*(const vect &v) const { return x * v.x
-		                                            + y * v.y; };
+		float operator*(const vect &v) const { return x * v.x + y * v.y; };
 
 		/* Equals? */
 		bool operator==(const vect &v) const { return x == v.x && y == v.y; };
 
 		/* Subtract from this vector */
-		void operator+=(const vect &v) { x += v.x; y += v.y; };
+		void operator+=(const vect &v)
+		{
+			x += v.x;
+			y += v.y;
+		};
 
 		/* Subtract from this vector */
-		void operator-=(const vect &v) { x -= v.x; y -= v.y; };
+		void operator-=(const vect &v)
+		{
+			x -= v.x;
+			y -= v.y;
+		};
 
 		/* Multiply into this vector by a scalar */
-		void operator*=(const float &s) { x *= s; y *= s; };
+		void operator*=(const float &s)
+		{
+			x *= s;
+			y *= s;
+		};
+
+		/* Entry-wise multiply into this vector by another vector */
+		void operator*=(const vect &v)
+		{
+			x *= v.x;
+			y *= v.y;
+		}
 
 		/* Divide into this vector by a scalar */
-		void operator/=(const float &s) { x /= s; y /= s; };
+		void operator/=(const float &s)
+		{
+			x /= s;
+			y /= s;
+		};
 
 		float x, y;
 	};
 
 	/* Multiply by a scalar */
-	inline vect operator*(const float &s, const vect &v) {
+	inline vect operator*(const float &s, const vect &v)
+	{
 		return vect(v.x * s, v.y * s);
 	};
 
+	inline vect zeroVector = vect(0, 0);
+
+	/**
+ 	 * Convert vector into a unit vector
```

Comment:
Example, if you feed in (1, 1) into normalize, you get out (rad 2 / 2, rad 2 / 2)!

Reply:
The reason for why we need a `normalize` is because, it is convenient to specify our direction as (0, 0), (0, 1), (1, 0), (0, -1), (-1, 0), (-1, 1), (1, 1), and (1, -1). Then, if we move a character by its direction vector, say (0, 1), it is only moving it one unit. However, if we move a character by (1, 1), that actually moves it rad(1^2 + 1^2) = rad(2) units. So, we must normalize the direction vector! After normalizing (1, 1), it becomes ((rad2)/2, (rad2)/2), and this vector's magnitude it 1.